noxfile.py

```diff
@@ -0,0 +1,147 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import annotations
+
+import nox
+
+nox.options.reuse_existing_virtualenvs = True
+
+
+def install(session: nox.Session, *args: str) -> None:
+    session.install(
+        "-v",
+        "-c",
+        "ci-constraints-requirements.txt",
+        *args,
+    )
+
+
+@nox.session
+@nox.session(name="tests-ssh")
+@nox.session(name="tests-randomorder")
+@nox.session(name="tests-nocoverage")
+def tests(session: nox.Session) -> None:
+    extras = "test"
+    if session.name == "tests-ssh":
+        extras += ",ssh"
+    if session.name == "tests-randomorder":
+        extras += ",test-randomorder"
+
+    install(session, f".[{extras}]")
+    install(session, "-e", "./vectors")
+
+    session.run("pip", "list")
+
+    if session.name != "tests-nocoverage":
+        session.run(
+            "pytest",
+            "-n",
+            "auto",
+            "--dist=worksteal",
+            "--cov=cryptography",
+            "--cov=tests",
+            "--durations=10",
+            *session.posargs,
+            "tests/",
+        )
+    else:
+        session.run(
+            "pytest",
+            "-n",
+            "auto",
+            "--dist=worksteal",
+            "--durations=10",
+            *session.posargs,
+            "tests/",
+        )
+
+
+@nox.session
+def docs(session: nox.Session) -> None:
+    install(session, ".[docs,docstest,sdist,ssh]")
+
+    temp_dir = session.create_tmp()
+    session.run(
+        "sphinx-build",
+        "-T",
+        "-W",
+        "-b",
+        "html",
+        "-d",
+        f"{temp_dir}/doctrees",
+        "docs",
+        "docs/_build/html",
+    )
+    session.run(
```

Comment:
Do we ever look at this output or publish it anywhere? If not, should we drop building this in CI?

Reply:
We broke it once, so we have this to catch regressions.

Reply:
RTD publishes the PDF AFAIK